### PR TITLE
Limit Data When Listing AppDefs and AppInstalls

### PR DESCRIPTION
**Pull request by SimonTheLeg**
Signed-off-by: Simon Bein <simontheleg@gmail.com>

**What this PR does / why we need it**:

This PR is migrated from https://github.com/kubermatic/kubermatic/pull/11005. Please see the original PR for all discussion

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubermatic/kubermatic/issues/10406

**What type of PR is this?**

**Special notes for your reviewer**:
/kind cleanup

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
- Limit Data When Listing AppDefs and AppInstalls
- Add GET endpoint for AppDefs
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/handler/v2/application_installation/conversion.go [hunk 1]
```diff
@@ -74,6 +74,47 @@ func convertInternalToAPIApplicationInstallation(in *appskubermaticv1.Applicatio
...
+			},
+			ApplicationRef: in.Spec.ApplicationRef,
+		},
+		Status: &apiv2.ApplicationInstallationListItemStatus{
+			Method:             in.Status.Method,
+			ApplicationVersion: in.Status.ApplicationVersion,
+		},
+	}
+
+	// TODO @vgramer decide if we want to extract this into a separate method, as duplicate code is used in convertInternalToAPIApplicationInstallation
```
Comment (vgramer) [inline pkg/handler/v2/application_installation/conversion.go:96]: yes would be nice to avoid duplicate code :)

**Comment by SimonTheLeg** [comment]
@vgramer all changes done. PR is ready for approval

**Review by vgramer — COMMENTED** [review]
/lgtm

File: pkg/api/v2/types.go [hunk 2]
```diff
@@ -1878,3 +1911,17 @@ type ClusterServiceAccount struct {
...
+// ApplicationDefinitionListItem is the object representing an ApplicationDefinitionListItem.
+// swagger:model ApplicationDefinitionListItem
+type ApplicationDefinitionListItem struct {
+	Name string `json:"name"`
+
+	Spec ApplicationDefinitionListItemSpec `json:"spec"`
+}
+
+// ApplicationDefinitionListItemSpec defines the desired state of ApplicationDefinitionListItemSpec.
+type ApplicationDefinitionListItemSpec struct {
```
Comment (pkprzekwas) [inline pkg/api/v2/types.go:1924]: I assume we expect this struct may grow over time, hence only one field for now.

**Pull request merged**